### ISSUE TITLE
Change PYSEC data format

### ISSUE THREAD

**Issue by donchkat** [google/osv.dev#1097]
Hi, I noticed a difference in the way the data is shown for GHSA and PYSEC sources.

The CVE-2021-32807 has two search results:

<img width="1888" alt="image" src="https://user-images.githubusercontent.com/74149435/223054436-0982a7a2-fe50-4739-8fa8-e2978587e4b5.png">

GHSA data is shown in separate tabs for each range:

<img width="742" alt="image" src="https://user-images.githubusercontent.com/74149435/223054899-bc67d65e-1b01-4068-99a6-a50f52083f8a.png">

<img width="695" alt="image" src="https://user-images.githubusercontent.com/74149435/223054942-320fdd67-9783-49ed-ba8f-e4327653f619.png">
which makes it much easier to consume the data correctly.

On the other half PYSEC data is not separated for tabs:
<img width="1132" alt="image" src="https://user-images.githubusercontent.com/74149435/223055123-7330a68c-ec09-46b3-b1a4-eb12d79b1752.png">

I have an automation that parses the data from OSV and creates yaml files for each CVE. The way the PYSEC data is served makes it harder to parse the data correctly - because I can't know for sure which range is relevant for which fixed version.

I was wandering if it's possible to change the way the PYSEC data is served to be separated to tabs as it is in GHSA data?

**Comment by oliverchang** [comment]
Hey @donchkat , I don't believe we've changed anything here -- PYSEC has always consolidated version ranges in the single affected entry. This enables entries to be a lot more concise and avoids some potential edge cases with version enumeration. 

May I understand a bit more on your use case here of knowing which version lands in which range?

**Comment by donchkat** [comment]
I'm not saying that something has change, I just noticed that when I'm parsing PYSEC json data, this is what I get:
<img width="535" alt="image" src="https://user-images.githubusercontent.com/74149435/223371437-adca394a-caf9-4915-9bc5-110106c797d9.png">

As you can see the affected versions list for two different ranges is the same. It happens because of the absence of separation between the different ranges.

While the GHSA json data parsing yields much better results:
<img width="344" alt="image" src="https://user-images.githubusercontent.com/74149435/223374489-2fe716f7-b041-4265-8464-88b2b801e096.png">

I just wanted to suggest an improvement to the structure of the PYSEC json data to make it easier to parse the data. (Similarly to the structure of the GHSA json that has separation between the ranges).

**Comment by oliverchang** [comment]
> I'm not saying that something has change, I just noticed that when I'm parsing PYSEC json data, this is what I get: <img alt="image" width="535" src="https://user-images.githubusercontent.com/74149435/223371437-adca394a-caf9-4915-9bc5-110106c797d9.png">
> 
> As you can see the affected versions list for two different ranges is the same. It happens because of the absence of separation between the different ranges.
> 
> While the GHSA json data parsing yields much better results: <img alt="image" width="344" src="https://user-images.githubusercontent.com/74149435/223374489-2fe716f7-b041-4265-8464-88b2b801e096.png">
> 
> I just wanted to suggest an improvement to the structure of the PYSEC json data to make it easier to parse the data. (Similarly to the structure of the GHSA json that has separation between the ranges).

Unfortunately we can't mandate our data sources to format things in certain ways. We generally recommend consolidating ranges in a single affected entry to avoid potential edge cases (see examples: https://ossf.github.io/osv-schema/#examples). The way OSV encodes version ranges is to consider them as part of a single "version timeline", and trying to separate them may introduce version encoding issues as we've seen in the past:

e.g. a producer may be inclined to produce:

```
affected: 
- ranges:
  - events:
    - introduced: 4.0
- ranges:
  - events:
    - introduced: 5.0
    - fixed: 5.4
 ```

Encoded like so, the OSV entry according to its [evaluation rules](https://ossf.github.io/osv-schema/#evaluation) will say everything from 4.0 onwards to the end of time is affected, (even 5.4, which is supposedly fixed). The intent here may have been to instead express that everything in 4.x is affected, and everything in 5.x is affected until 5.4. 

The more correct way to encode this situation would be:

```
affected: 
- ranges:
  - events:
    - introduced: 4.0
    - introduced: 5.0
    - fixed: 5.4
 ```

Which does not have such problems.
